src/parse/validator.rs

```diff
@@ -250,131 +257,122 @@ impl<'help, 'app, 'parser> Validator<'help, 'app, 'parser> {
-            // it doesn't conflict with anything
-            //
-            // TODO: @refactor Do it in a more elegant way
-            if matcher
-                .get(name)
-                .map_or(false, |a| a.ty == ValueType::DefaultValue)
-            {
-                debug!("Validator::gather_conflicts:iter: This is default value, skipping.",);
-                continue;
-            }
-
-            if let Some(arg) = self.p.app.find(name) {
-                // Since an arg was used, every arg it conflicts with is added to the conflicts
-                for conf in &arg.blacklist {
-                    if self.p.app.find(conf).is_some() {
-                        if conf != name {
-                            self.c.insert(conf.clone());
-                        }
-                    } else {
-                        // for g_arg in self.p.app.unroll_args_in_group(conf) {
-                        //     if &g_arg != name {
-                        self.c.insert(conf.clone()); // TODO ERROR is here - groups allow one arg but this line disallows all group args
-                                                     //     }
-                                                     // }
-                    }
+        matcher
+            .arg_names()
+            .filter(|name| {
+                debug!("Validator::gather_conflicts:iter: id={:?}", name);
+                // if arg is "present" only because it got default value
+                // it doesn't conflict with anything and should be skipped
+                let skip = matcher
+                    .get(name)
+                    .map_or(false, |a| a.ty == ValueType::DefaultValue);
+                if skip {
+                    debug!("Validator::gather_conflicts:iter: This is default value, skipping.",);
                 }
-
-                // Now we need to know which groups this arg was a member of, to add all other
-                // args in that group to the conflicts, as well as any args those args conflict
-                // with
-
-                for grp in self.p.app.groups_for_arg(&name) {
-                    if let Some(g) = self
-                        .p
-                        .app
-                        .groups
-                        .iter()
-                        .find(|g| !g.multiple && g.id == grp)
-                    {
-                        // for g_arg in self.p.app.unroll_args_in_group(&g.name) {
-                        //     if &g_arg != name {
-                        self.c.insert(g.id.clone());
-                        //     }
-                        // }
+                !skip
+            })
+            .for_each(|name| {
+                if let Some(arg) = self.p.app.find(name) {
+                    // Since an arg was used, every arg it conflicts with is added to the conflicts
+                    for conf in &arg.blacklist {
+                        self.c.insert(conf.clone());
```

Comment:
Can you please copy the comment that was here from above?

Reply:
Done, using multiline comment since the current way of commenting does not seem to play well with `rustfmt`.